utils.go

```diff
@@ -16,18 +17,28 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	// First, look for a PROPAGATION_FORMAT_GO_HTTP_HEADER injector (our preference).
+	injector := sp.PropagationInjectorForFormat(PROPAGATION_FORMAT_GO_HTTP_HEADER)
```

Comment:
I missed this comment. sry.

The implementation looks good.

Like i've said before I'd rather see a clean separation of how a span is put on a protocol's wire from the particular framework at hand by re-using the same API and using a delegation approach.

That is the notion of a "carrier" becomes the same as the "interface" in the Injector/Extractor.

So in this situation you have the delegation model of:

    HttpHeaderSpanInjector --> SplitTextInjector
This is easily done because the interaction between the two is a `map<string,string>` and is dictated by the latter. 

> the formats declared within OpenTracing proper are mandatory

Agreed. This can be done via `SplitTextInjector`, rather than adding the Carrier API concept.

In addition the notion of the traceState and traceAttributes does not need to be explicitly represented in the API. It is represented rather by the implementation of SplitTextInjector (and BinaryInjector).